### PR TITLE
fix: clarify PUT ensure semantics

### PR DESCRIPTION
**Pull request by shikhar**
## Summary
- Change hidden basin/stream ensure paths to use full desired configs for PUT, applying defaults before comparing or storing existing resources.
- Replace create-or-reconfigure plumbing with `ProvisionMode` and `ProvisionResult::{Created, Updated}` across common, lite backend, SDK, handlers, and CLI apply.
- Update `s2 apply --dry-run` to diff effective desired state, including stream config merged with basin defaults, and report existing resources as updated rather than reconfigured.
- Add coverage for ensure default resets, idempotency behavior, active streamer updates, and API conversion edge cases for omitted nested config fields.

## Tests
- cargo check -q
- just fmt
- just test

## Notes
- api/specs submodule generated diff is intentionally not included in this PR.

### CODE REVIEW

File: lite/src/backend/streams.rs [hunk 1]
```diff
@@ -140,124 +136,119 @@ impl Backend {
...
+            txn.put(&stream_meta_key, kv::stream_meta::ser_value(meta))?;
+            let stream_id = StreamId::new(&basin, &stream);
+            if matches!(&outcome, ProvisionResult::Created(_)) {
+                txn.put(
+                    kv::stream_id_mapping::ser_key(stream_id),
+                    kv::stream_id_mapping::ser_value(&basin, &stream),
+                )?;
+                let created_secs = meta.created_at.unix_timestamp();
+                let created_secs = if created_secs <= 0 {
+                    0
```
[the remark this answers is not in the document]
Comment (shikhar) [reply lite/src/backend/streams.rs:202]: pre-existing semantic

**Comment by shikhar** [comment]
@greptileai recheck
